File: Mntone.SplatoonClient/Mntone.SplatoonClient.Entities/RankingUser.cs
using Mntone.SplatoonClient.Entities.Internal;
using System;
using System.Runtime.Serialization;

namespace Mntone.SplatoonClient.Entities
{
	[DataContract]
	[System.Diagnostics.DebuggerDisplay("{this.Rank}: {this.Name} ({this.Score}) ")]
	public sealed class RankingUser
	{
		internal RankingUser(string id, uint rank, uint score, string name, Uri miiImageUri,
			Uri weaponImageUri, Uri weaponRetinaImageUri,
			Uri headImageUri, Uri headRetinaImageUri,
			Uri clothesImageUri, Uri clothesRetinaImageUri,
			Uri shoesImageUri, Uri shoesRetinaImageUri)
		{
			this.ID = id;
			this.Rank = rank;
			this.Score = score;
			this.Name = name;
			this.MiiImageUri = miiImageUri;
			this.WeaponImageUri = weaponImageUri;
			this.WeaponRetinaImageUri = weaponRetinaImageUri;
			this.HeadImageUri = headImageUri;
			this.HeadRetinaImageUri = headRetinaImageUri;
			this.ClothesImageUri = clothesImageUri;
			this.ClothesRetinaImageUri = clothesRetinaImageUri;
			this.ShoesImageUri = shoesImageUri;
			this.ShoesRetinaImageUri = shoesRetinaImageUri;
		}

		/// <summary>
		/// User id
		/// </summary>
		[DataMember(Name = "hashed_id")]
		public string ID { get; private set; }

		/// <summary>
		/// Rank
		/// </summary>
		public uint Rank { get; private set; }

		[DataMember(Name = "rank", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private uint[] RankImpl
		{
			get { return this.Rank.ToUInt32Array(); }
			set { this.Rank = value.ToUInt32(); }
		}

		/// <summary>
		/// Score
		/// </summary>
		public uint Score { get; private set; }

		[DataMember(Name = "score", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private uint[] ScoreImpl
		{
			get { return this.Score.ToUInt32Array(); }
			set { this.Score = value.ToUInt32(); }
		}

		/// <summary>
		/// Name
		/// </summary>
		[DataMember(Name = "mii_name", IsRequired = true)]
		public string Name { get; private set; }

		/// <summary>
		/// Mii iamge uri
		/// </summary>
		[DataMember(Name = "mii_url", IsRequired = true)]
		public Uri MiiImageUri { get; private set; }

		/// <summary>
		/// Weapon image uri
		/// </summary>
		public Uri WeaponImageUri { get; private set; }

		[DataMember(Name = "weapon", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string WeaponImageUriImpl
		{
			get { return this.WeaponImageUri.ToFilename(); }
			set { this.WeaponImageUri = value.ToWeaponOrGearImageUri("weapon", false); }
		}

		/// <summary>
		/// Weapon retina image uri
		/// </summary>
		public Uri WeaponRetinaImageUri { get; private set; }

		[DataMember(Name = "weapon2x", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string WeaponRetinaImageUriImpl
		{
			get { return this.WeaponRetinaImageUri.ToFilename(); }
			set { this.WeaponRetinaImageUri = value.ToWeaponOrGearImageUri("weapon", true); }
		}

		/// <summary>
		/// Head image uri
		/// </summary>
		public Uri HeadImageUri { get; private set; }

		[DataMember(Name = "head", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string HeadImageUriImpl
		{
			get { return this.HeadImageUri.ToFilename(); }
			set { this.HeadImageUri = value.ToWeaponOrGearImageUri("head", false); }
		}

		/// <summary>
		/// Weapon retina image uri
		/// </summary>
		public Uri HeadRetinaImageUri { get; private set; }

		[DataMember(Name = "head2x", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string HeadRetinaImageUriImpl
		{
			get { return this.HeadRetinaImageUri.ToFilename(); }
			set { this.HeadRetinaImageUri = value.ToWeaponOrGearImageUri("head", true); }
		}

		/// <summary>
		/// Clothes image uri
		/// </summary>
		public Uri ClothesImageUri { get; private set; }

		[DataMember(Name = "clothes", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string ClothesImageUriImpl
		{
			get { return this.ClothesImageUri.ToFilename(); }
			set { this.ClothesImageUri = value.ToWeaponOrGearImageUri("clothes", false); }
		}

		/// <summary>
		/// Weapon retina image uri
		/// </summary>
		public Uri ClothesRetinaImageUri { get; private set; }

		[DataMember(Name = "clothes2x", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string ClothesRetinaImageUriImpl
		{
			get { return this.ClothesRetinaImageUri.ToFilename(); }
			set { this.ClothesRetinaImageUri = value.ToWeaponOrGearImageUri("clothes", true); }
		}

		/// <summary>
		/// Shoes image uri
		/// </summary>
		public Uri ShoesImageUri { get; private set; }

		[DataMember(Name = "shoes", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string ShoesImageUriImpl
		{
			get { return this.ShoesImageUri.ToFilename(); }
			set { this.ShoesImageUri = value.ToWeaponOrGearImageUri("shoes", false); }
		}

		/// <summary>
		/// Weapon retina image uri
		/// </summary>
		public Uri ShoesRetinaImageUri { get; private set; }

		[DataMember(Name = "shoes2x", IsRequired = true)]
		[System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
		private string ShoesRetinaImageUriImpl
		{
			get { return this.ShoesRetinaImageUri.ToFilename(); }
			set { this.ShoesRetinaImageUri = value.ToWeaponOrGearImageUri("shoes", true); }
		}
	}
}

File: net.mntone.splatoonclient/net.mntone.splatoonclient.entities/src/main/java/net/mntone/splatoonclient/entities/RankingUser.java
package net.mntone.splatoonclient.entities;

import org.json.JSONObject;

import java.net.MalformedURLException;
import java.net.URL;

public final class RankingUser
{
	private final String _id;
	private final int _rank;
	private final int _score;
	private final String _name;
	private final URL _miiImageUri;
	private final URL _weaponImageUri;
	private final URL _weaponRetinaImageUri;
	private final URL _headImageUri;
	private final URL _headRetinaImageUri;
	private final URL _clothesImageUri;
	private final URL _clothesRetinaImageUri;
	private final URL _shoesImageUri;
	private final URL _shoesRetinaImageUri;

	RankingUser(final String id, final int rank, final int score, final String name, final URL miiImageUri, final URL weaponImageUri, final URL weaponRetinaImageUri, final URL headImageUri, final URL headRetinaImageUri, final URL clothesImageUri, final URL clothesRetinaImageUri, final URL shoesImageUri, final URL shoesRetinaImageUri)
	{
		this._id = id;
		this._rank = rank;
		this._score = score;
		this._name = name;
		this._miiImageUri = miiImageUri;
		this._weaponImageUri = weaponImageUri;
		this._weaponRetinaImageUri = weaponRetinaImageUri;
		this._headImageUri = headImageUri;
		this._headRetinaImageUri = headRetinaImageUri;
		this._clothesImageUri = clothesImageUri;
		this._clothesRetinaImageUri = clothesRetinaImageUri;
		this._shoesImageUri = shoesImageUri;
		this._shoesRetinaImageUri = shoesRetinaImageUri;
	}

	RankingUser(final JSONObject json) throws MalformedURLException
	{
		this._id = json.getString("hashed_id");
		this._rank = JsonUtil.convertJsonIntArrayToInt(json.getJSONArray("rank"));
		this._score = JsonUtil.convertJsonIntArrayToInt(json.getJSONArray("score"));
		this._name = json.getString("mii_name");
		this._miiImageUri = new URL(json.getString("mii_url"));
		this._weaponImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("weapon"), "weapon", false);
		this._weaponRetinaImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("weapon2x"), "weapon", true);
		this._headImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("head"), "head", false);
		this._headRetinaImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("head2x"), "head", true);
		this._clothesImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("clothes"), "clothes", false);
		this._clothesRetinaImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("clothes2x"), "clothes", true);
		this._shoesImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("shoes"), "shoes", false);
		this._shoesRetinaImageUri = JsonUtil.toWeaponOrGearImageUri(json.getString("shoes2x"), "shoes", true);
	}

	public String getId()
	{
		return this._id;
	}

	public int getRank()
	{
		return this._rank;
	}

	public int getScore()
	{
		return this._score;
	}

	public String getName()
	{
		return this._name;
	}

	public URL getMiiImageUri()
	{
		return this._miiImageUri;
	}

	public URL getWeaponImageUri()
	{
		return this._weaponImageUri;
	}

	public URL getWeaponRetinaImageUri()
	{
		return this._weaponRetinaImageUri;
	}

	public URL getHeadImageUri()
	{
		return this._headImageUri;
	}

	public URL getHeadRetinaImageUri()
	{
		return this._headRetinaImageUri;
	}

	public URL getClothesImageUri()
	{
		return this._clothesImageUri;
	}

	public URL getClothesRetinaImageUri()
	{
		return this._clothesRetinaImageUri;
	}

	public URL getShoesImageUri()
	{
		return this._shoesImageUri;
	}

	public URL getShoesRetinaImageUri()
	{
		return this._shoesRetinaImageUri;
	}
}